src/openzaak/config/bootstrap/authorizations.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2024 Dimpact
+import logging
+import sys
+from contextlib import contextmanager
+from io import StringIO
+
+from django.conf import settings
+from django.core.management import call_command
+from django.core.serializers.base import DeserializationError
+from django.db import transaction
+
+import yaml
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import ConfigurationRunFailed
+from vng_api_common.authorizations.models import Autorisatie
+from vng_api_common.models import JWTSecret
+
+from openzaak.components.autorisaties.models import Applicatie, CatalogusAutorisatie
+
+logger = logging.getLogger(__name__)
+
+
+@contextmanager
+def override_stdin(data):
+    original_stdin = sys.stdin
+    sys.stdin = data
+    try:
+        yield
+    finally:
+        sys.stdin = original_stdin
+
+
+def delete_existing_configuration() -> None:
+    logger.info("Removing all existing authorization configuration")
+
+    Applicatie.objects.all().delete()
+    Autorisatie.objects.all().delete()
+    CatalogusAutorisatie.objects.all().delete()
+    JWTSecret.objects.all().delete()
+
+
+class AuthorizationConfigurationStep(BaseConfigurationStep):
+    """
+    Configure the Applicaties and Autorisaties.
+
+    From: https://open-zaak.readthedocs.io/en/stable/manual/api-authorizations.html
+    """
+
+    verbose_name = "Authorization configuration"
+    required_settings = [
+        "AUTHORIZATIONS_CONFIG_FIXTURE_PATH",
+    ]
+    enable_setting = "AUTHORIZATIONS_CONFIG_ENABLE"
+
+    def is_configured(self) -> bool:
+        with open(settings.AUTHORIZATIONS_CONFIG_FIXTURE_PATH) as f:
+            data = yaml.safe_load(f.read())
+            existing_uuids = set(
+                str(uuid) for uuid in Applicatie.objects.values_list("uuid", flat=True)
+            )
+            applicatie_uuids = {
+                row["fields"]["uuid"]
+                for row in data
+                if row["model"] == "authorizations.applicatie"
+            }
+            applicaties_configured = applicatie_uuids.issubset(existing_uuids)
+
+            existing_identifiers = set(
+                str(uuid)
+                for uuid in JWTSecret.objects.values_list("identifier", flat=True)
+            )
+
+            identifiers = {
+                row["fields"]["identifier"]
+                for row in data
+                if row["model"] == "vng_api_common.jwtsecret"
+            }
+            secrets_configured = identifiers.issubset(existing_identifiers)
+
+        return applicaties_configured and secrets_configured
+
+    @transaction.atomic
+    def configure(self):
+        if settings.AUTHORIZATIONS_CONFIG_DELETE_EXISTING:
+            delete_existing_configuration()
```

Comment:
@annashamray I added this to allow using the fixture as a single source of truth, but I just realized that other ConfigurationSteps also add applicaties and jwtsecrets. How should I treat those? I guess I could exclude the stuff created in previous steps from the .delete, but I wanted to check with you before I make it more complicated

Reply:
It can be a leaky abstraction. but I'd check if other steps which add autorisation data are enabled. 
If they are enabled and AUTHORIZATIONS_CONFIG_DELETE_EXISTING = True, I'd raise an Exception

It's first time then we have configuration steps that can't be used together. Maybe in the future we can add some steprelation validator that can check which will have some mapping of steps that can't be run together?

Reply:
I added code to raise an exception if any of OPENZAAK_NOTIF_CONFIG_ENABLE, NOTIF_OPENZAAK_CONFIG_ENABLE, DEMO_CONFIG_ENABLE are True